README.md

```diff
@@ -1,32 +1,279 @@
 
-The process currently hard-wires `kdiff3` for merging and conflict resolution.
-`kdiff3` runs in auto mode and will only pop up a UI if conflict resolution is
-not possible (note: X display needed in this case). There are two ways to invoke
-code synchronizer:
+- linux-aarch_64
+- linux-x86_32
+- linux-x86_64
+- osx-x86_32
+- osx-x86_64
+- win32
 
-1. Invoking directly through a gradle task (see the build.gradle:runSynchronizer
-   task).
-2. Invoking a code generation pipeline which contains synchronization step
-   through [artman](https://github.com/googleapis/artman/#usage).
+More platforms are likely supported when building protoc from C++ source.
+
+## Usage
+
+There are two main ways to generate clients:
+
+1. Build from source and run directly. This is detailed below.
+2. Invoke a code generation pipeline through
+   [artman]( https://github.com/googleapis/artman/#usage). This takes care of several of the steps
+   below and is not documented here.
+
+### Prerequisites for code generation
+
+1. You need a proto file describing your API. Proto files of existing Google APIs are available at
+[googleapis](https://github.com/googleapis/googleapis).
+  - Example: https://github.com/googleapis/googleapis/blob/master/google/cloud/language/v1/language_service.proto
+2. You need a yaml file describing certain service values. This is not yet well-documented, but you can
+  figure out some of the settings by looking at existing files.
+  - Example: https://github.com/googleapis/googleapis/blob/master/google/cloud/language/language_v1.yaml
+3. An active internet connection. The first time you build toolkit, it will download Java dependencies over
+  the network.
+
+### Process overview for generating a client library
+
+1. Generate a descriptor file from the proto (once per API)
+2. Run client config generation (once per API)
+3. Manually tweak the generated client config
+4. Run code generation (once per API x language combination of interest)
+
+Each of these steps are described in more detail below. Note: the instructions assume you are running on a
+Unix-y system; if you are using Windows, you will need to tweak the steps for yourself.
+
+### Set up prerequisites for building from source and running directly
+
+Firstly, make sure you have the
+[Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) installed.
+
+Next, you need to install `protoc` if you don't have it yet: see
+[protocol-compiler-installation](https://github.com/google/protobuf#protocol-compiler-installation).
+The pre-built binaries are the easiest. There aren't instructions for installation;
+you just have to download the correct binaries for your platform and put them on your path.
+Given a platform ${PROTOC_PLATFORM} and a protoc version ${PROTOC_VERSION}:
+
+1. Download the latest zip from https://github.com/google/protobuf/releases ending in "-${PROTOC_PLATFORM}.zip"
```

Comment:
`the latest zip file ending in "..." from https://....`

Reply:
Will do.